git_theta/utils.py

```diff
@@ -73,14 +73,15 @@ def __get__(self, obj, objtype=None):
 class EnvVarConstants:
     CHECKPOINT_TYPE = EnvVar(name="GIT_THETA_CHECKPOINT_TYPE", default="pytorch")
     UPDATE_TYPE = EnvVar(name="GIT_THETA_UPDATE_TYPE", default="dense")
-    UPDATE_DATA_PATH = EnvVar(name="GIT_THETA_UPDATE_DATA_PATH", default="update.pt")
+    UPDATE_DATA_PATH = EnvVar(name="GIT_THETA_UPDATE_DATA_PATH", default="")
```

Comment:
Why doesn't this get a default anymore?

Reply:
Basically, when an update is done with side loaded information (like a sparse or lora update), the on-disk/working checkpoint doesn't change, but the checkout in the git history does (it has the side loaded information in it). This causes false positives when you run `git status`. As such, we include a check for side-loaded information in some of our scripts that will write the updated version to disk so that git doesn't detect a diff.

With the default value, git-theta was always triggering that extra write, even in a dense update where the result is overwriting the model with the same values.


So removing this default value skips that extra smudge in cases where there isn't any side-loaded information.